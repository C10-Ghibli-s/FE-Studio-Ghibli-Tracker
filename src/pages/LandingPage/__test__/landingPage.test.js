/**
 * @jest-environment jsdom
 */
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import testRenderer from "react-test-renderer";
import { LandingPage } from '../index';

describe('Landing Page', () => {
    test('testing the snapshot of the CTA buttons', () => {
        const landingPage = testRenderer.create(<LandingPage/>).toJSON();
        expect(landingPage).toMatchSnapshot();
    });
    test('landing page renders all its elements', () => {
        render(<LandingPage/>);
        const CTA_buttons = screen.getAllByRole('link');
        const landingImage = screen.getAllByRole('img');
        CTA_buttons.map(button => expect(button).toBeInTheDocument());
        landingImage.map(image => expect(image).toBeInTheDocument());
    })
});