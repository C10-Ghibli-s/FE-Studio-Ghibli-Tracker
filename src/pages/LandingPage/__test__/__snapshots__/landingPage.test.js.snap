// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page testing the snapshot of the CTA buttons 1`] = `
Array [
  <section
    className="login"
  >
    <p
      className="login__text"
    >
      have already an account?
    </p>
    <a
      className="loginButton"
      href="/login"
    >
      Login
    </a>
  </section>,
  <div
    className="landingContainer"
  >
    <section
      className="landingPage__hero"
    >
      <div
        className="appLogo"
      >
        <picture
          className="secondaryMarkLogotype"
        >
          <img
            alt="Studio Ghibli Logotype"
            src="test-file-stub"
          />
        </picture>
        <picture
          className="trackerLogotype"
        >
          <img
            src="test-file-stub"
          />
        </picture>
      </div>
      <picture>
        <img
          alt="Studio Ghibli Works Catalogue"
          className="landingPage__Image"
          src="test-file-stub"
        />
      </picture>
    </section>
    <section
      className="landingDescription"
    >
      <article
        className="landingDescription__text"
      >
        <h2>
           
          <span
            className="landing__text--search"
          >
            Search
          </span>
          , 
          <span
            className="landing__text--discover"
          >
            Discover
          </span>
          , 
          <span
            className="landing__text--rate"
          >
            Rate
          </span>
        </h2>
        <div>
          <p>
            With Studio Ghibli 
            <span
              className="std_ghibli_tracker"
            >
              Tracker
            </span>
             you can:
          </p>
          <p>
            <span
              className="landing__text--search"
            >
              Search
            </span>
             all the animated masterpieces produced by Studio Ghibli,
          </p>
          <p>
            <span
              className="landing__text--discover"
            >
              Discover
            </span>
             the incredible and fascinating catalogue of memorable movies,
          </p>
          <p>
            and 
            <span
              className="landing__text--rate"
            >
              Rate
            </span>
             them to share with everyone your love and thoughts for your favorite movies.
          </p>
        </div>
      </article>
      <a
        className="registerNow"
        href="/register"
      >
         Register Now!
      </a>
    </section>
  </div>,
]
`;
